app/routes/post.py

```diff
@@ -1,42 +1,49 @@
 from app.server import server
 from app.helpers.render import render_template, render_json
-from app.helpers.markdown import render_markdown_safe
+from app.helpers import tasks
 from app.controllers import post
 from flask import request, redirect, url_for, g, abort
 
+
 @server.route("/posts")
 def get_posts():
     try:
-        page = int(request.args.get('p') or 1)
-    except:
+        page = int(request.args.get('p', 1))
+    except ValueError:
         return abort(400)
-    
+
     return render_template('posts.html', posts=post.get_posts(page=page))
 
+
 @server.route("/post/<int:post_id>")
 def get_post(post_id):
     matched_post = post.get_post(post_id=post_id)
-    if matched_post is None: return abort(404)
-    
-    body = render_markdown_safe(matched_post.body, id="P" + str(matched_post.id))
-    if body is None: return abort(500)
-    
+    if matched_post is None:
+        return abort(404)
+
+    # TODO: async loading
+    body = tasks.render_markdown.delay(matched_post.body, id="P" + str(matched_post.id)).wait()
```

Comment:
Python throws an error that `id` param shouldn't be here. Removing it fixes (this) problem.